examples/benchmark/datagen.py

```diff
@@ -0,0 +1,205 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+import abc
+import os
+import tarfile
+import tempfile
+from pathlib import Path
+from typing import Generator
+
+import boto3
+import click
+import numpy as np
+import prefixed
+import prefixed as pr
+import yaml
+from PIL import Image
+from joblib import parallel_config, Parallel, delayed, cpu_count
+from tqdm import tqdm
+
+
+class DataGenerator(abc.ABC):
+    @abc.abstractmethod
+    def generate(self, index: int, destdir: Path) -> bytes:
+        pass
+
+
+class ImageGenerator(DataGenerator):
+    def __init__(self, width: int, height: int, img_format: str = "JPEG"):
+        self.width = width
+        self.height = height
+        self.img_format = img_format
+
+    def generate(self, index: int, destdir: Path):
+        img = Image.fromarray(
+            np.random.randint(
+                0, high=256, size=(self.height, self.width, 3), dtype=np.uint8
+            )
+        )
+        img_file: Path = destdir / f"image_{index}.{self.img_format}"
+        img.save(img_file)
+
+
+class Utils:
+    @staticmethod
+    def batch_files(
+        files: list[Path], size_threshold: float
+    ) -> Generator[list[Path], None, None]:
+        group: list[Path] = []
+        total = 0
+        for file in files:
+            new_total = total + file.stat().st_size
+            if new_total > size_threshold:
+                yield group
+                group = [file]
+                total = file.stat().st_size
+            else:
+                group.append(file)
+                total = new_total
+        if len(group) > 0:
+            yield group
+
+    @staticmethod
+    def tar_files(name: str, workdir: Path, files: list[Path]) -> Path:
+        tar_path: Path = workdir / f"{name}.tar"
+        with tarfile.TarFile(tar_path, mode="w") as tarball:
+            for file in files:
+                tarball.add(file, arcname=file.name)
+                os.remove(file)
+        return tar_path
+
+    @staticmethod
+    def parse_resolution(
+        ctx: click.Context, param: str, value: str
+    ) -> tuple[int, int] | None:
+        if value is not None:
+            width, _, height = value.replace(" ", "").lower().partition("x")
+            return int(width), int(height)
+
+    @staticmethod
+    def upload_to_s3(region: str, file: Path, bucket: str, prefix: str | None):
+        s3_client = boto3.client("s3", region_name=region)
+        s3_key_parts = []
+        if prefix:
+            s3_key_parts.append(prefix)
+        s3_key_parts.append(os.path.basename(file))
+        s3_key = "/".join(s3_key_parts)
+        s3_client.upload_file(file, bucket, s3_key)
+
+    @staticmethod
+    def parse_hr_bytes(ctx: click.Context, param: str, value: str) -> float | None:
```

Comment:
What does `hr` stand for?

Reply:
Sorry for the obscure shortform. `hr == human readable` (oh the irony!)